src/persist-client/src/cli/inspect.rs

```diff
@@ -254,19 +273,50 @@ pub async fn fetch_latest_state_rollup(
     let blob = BlobConfig::try_from(blob_uri).await?;
     let blob = blob.clone().open().await?;
 
-    if let Some(diff_buf) = consensus.head(&shard_id.to_string()).await? {
-        let diff = ProtoStateDiff::decode(diff_buf.data).expect("invalid encoded diff");
-        let rollup_key = PartialRollupKey(diff.latest_rollup_key);
-        let rollup_buf = blob
-            .get(&rollup_key.complete(&shard_id))
-            .await
-            .unwrap()
-            .unwrap();
-        let proto = ProtoStateRollup::decode(rollup_buf.as_slice()).expect("invalid encoded state");
-        return Ok(proto);
-    }
+    let rollup_key = if let Some(seqno) = seqno {
+        if let Some(rollup) = rollup {
+            PartialRollupKey::new(seqno, &rollup)
+        } else {
```

Comment:
This branch (seqno specified but no rollup key) is a bit complicated. Is it also useful?

Reply:
In the recent incident it would not have helped -- since it can be quite expensive to discover the rollup id when the initial state or number of diffs is large -- but it feels like it would be annoying to have to hunt down a rollup id for an ordinary shard.

Reply:
off the top of my head, I _think_ any time you'd come in knowing a rollup seqno, you'd actually have the full `PartialRollupKey` (seqno + rollupid), so it's probably most straightforward to have a single string flag that is just turned directly into a `PartialRollupKey`